internal/npm_install/npm_install.bzl

```diff
@@ -277,6 +310,7 @@ cd "{root}" && "{npm}" {npm_args}
     )
     if result.return_code:
         fail("pre_process_package_json.js failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (result.stdout, result.stderr))
+    is_pnp = json_parse(result.stdout)["pnp"]
```

Comment:
Since you can parse json, why not parse `package.json` directly, rather than going through the script?

Reply:
Because Starlark doesn't support file i/o as far as I know

Reply:
Try https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#read.

Reply:
oh nice, makes sense it's exposed on repository rules